src/utils/parsers/settings.js

```diff
@@ -0,0 +1,71 @@
+import s from "underscore.string";
+
+const fortranDoubleRegex =
+    "([-+]?" + // Optional leading sign
+    "\\d*" + // Zero or more digits before the decimal point
+    "\\.?" + // Optional decimal point
+    "\\d*" + // Zero or more digits after the decimal point
+    "(?:[Eed][+-]?\\d+)?" + // Optional exponent part
+    ")";
+
+const fortranNamelistRegex =
+    "&" + // Start with an ampersand
+    "%s" + // Namelist name
+    "((?:\\s|\\S)*?)" + // Matches any sequence of space or non-space characters
+    "\\/"; // Ends with a slash
+
+const fortranCardsRegex =
+    "^\\s*\\/" + // Slash at the start of a line with any leading spaces
+    "(?![\\s\\S]*^\\/)" + // Negative lookahead for a slash at the beginning of the next line
+    "([\\s\\S]*)"; // Capture all characters till end
+
+const keyValueRegex =
+    "^\\s*" + // Key name at the start of a line with any leading spaces
+    "%s" + // Key name
+    "\\s*=\\s*" + // Equal sign with any leading and trailing spaces
+    "%s" + // Value
+    "\\s*" +
+    "\\n"; // Ends with a newline character
+
+const stringRegex = "([+\\w.\\-\\/]*)"; // Matches alphanumeric, plus, period, hyphen, and slash characters
+
+const fortranStringRegex =
+    "'" + // Starting single quote
+    "([\\w.\\-\\+\\/ ]*)" + // Matches alphanumeric, period, hyphen, plus, slash, and space characters
+    "'"; // Ending single quote
+
+const fortranArrayRegex =
```

Comment:
It seems the namelist format supports a few more options for representing arrays. According to [ref1](https://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vnc6/index.html#:~:text=You%20can%20alternatively,program%20segment%20above%3A) and [ref2](https://www.cenapad.unicamp.br/parque/manuais/Xlf/lr96.HTM), they could also take the following form:
```
&FOO
A(3) = 2, 10, 15, 16
mat(2, 2) = 2.2
/
```
Although unlikely for Quantum Espresso, we should ensure the array regexes include such cases.